packages/backend/src/model/Project.ts

```diff
@@ -117,6 +129,47 @@ function toBackendTrackedTxsConfig(
   }
 }
 
+const chainConverter = new ChainConverter(
+  chains.map((x) => ({ name: x.name, chainId: ChainId(x.chainId) })),
+)
+
+export function layer3ToProject(layer3: Layer3): Project {
+  return {
+    projectId: layer3.id,
+    slug: layer3.display.slug,
+    type: 'layer3',
+    isUpcoming: layer3.isUpcoming,
+    escrows: layer3.config.escrows.map((escrow) => {
+      const chain = escrow.chain
+      assert(chain, ` ${layer3.id}: chain is required for L3 escrow`)
+      const chainId = chainConverter.toChainId(chain)
+
+      const tokensOnChain = tokenList.filter((t) => t.chainId === chainId)
+
+      return {
+        address: escrow.address,
+        sinceTimestamp: escrow.sinceTimestamp,
+        tokens:
+          escrow.tokens === '*'
+            ? tokensOnChain
+            : escrow.tokens
+                .filter((t) => t !== 'ETH')
+                .map((tokenSymbol) => {
+                  const token = tokensOnChain.find(
+                    (t) => t.symbol === tokenSymbol,
+                  )
+                  assert(
+                    token,
+                    `${layer3.id}: token with symbol ${tokenSymbol} not found on ${chain}`,
+                  )
+                  return token
+                }),
```

Comment:
Please make those calculations higher in the tree. It's very hard to see what's going on with this little space in the line left.